### PR TITLE
Pass if score equals fail_under

### PR DESCRIPTION
**Pull request by chohner**
## Steps

- [ ] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [ ] If it's a new feature or an important bug fix, add a What's New entry in `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Description
Checks whether score of a run is larger or equals `config.fail_under`. This not only semantically makes sense but also allows for setting `fail_under` to `10.0` and still pass on a perfect score.

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :bug: Bug fix  |

### CODE REVIEW

File: tests/test_self.py [hunk 1]
```diff
@@ -665,7 +665,7 @@ def test_fail_under(self):
                 "--enable=all",
                 join(HERE, "regrtest_data", "fail_under_plus6.py"),
             ],
-            code=0,
+            code=16,
```
Comment (Pierre-Sassoulas) [inline tests/test_self.py:668]: Why the change here ?

**Review by Pierre-Sassoulas — APPROVED** [review]
I have a small question but thank you, this makes a lot of sense !

File: tests/test_self.py (hunk 1 above)
Comment (chohner) [reply tests/test_self.py:668]: Good point, I think I just saw that this was exactly testing `score == config.fail_under` and expected that I would now have to change the return value. Reverting now, but why wasn't this failing before?

Comment (Pierre-Sassoulas) [reply tests/test_self.py]: I'm not super familiar with this part of the code. Would you mind investigating a little ? Maybe you'll kill two bugs with one merge-request 😄

Comment (chohner) [reply tests/test_self.py]: This took a bit longer than expected but I think I fixed & expanded all regression tests in 376b3d0

The returned tests scores were different from the comment and I had to modify the regression test files a bit to get a stable score to test against. I covered all over, exact and under cases for positive and negative numbers

Comment (Pierre-Sassoulas) [reply tests/test_self.py]: Thanks a lot :)

**Comment by Pierre-Sassoulas** [comment]
Thank you for the change and congratulation on becoming a pylint contributor :)
